src/MongoDB.Driver.Core/Core/Operations/DeleteOpcodeOperation.cs

```diff
@@ -26,6 +26,7 @@ namespace MongoDB.Driver.Core.Operations
     /// <summary>
     /// Represents a delete operation using the delete opcode.
     /// </summary>
+    [Obsolete("Use BulkMixedWriteOperation instead.")]
```

Comment:
Let's consider whether we can just delete this class (and the others like it), despite the fact that it's public.

Reply:
@rstam do you now why that is?  Is it just that we never updated Legacy to use write commands?

Is it worth updating (in scope of another ticket) so that we can remove all this otherwise-dead code?

Reply:
We did update the Legacy to use write commands.

If write command are supported the DeleteOpcodeOperation will use a DeleteOpcodeOperationEmulator to use write commands while providing the DeleteOpcodeOperation interface and result type to the Legacy calling code.

Reply:
The Legacy code could instantiate a DeleteOpcodeOperationEmulator directly if we want to assume that write commands are supported.

Reply:
There is still the check on `writeConcern.IsAcknowledged` to consider. As I understand it, this class will continue to use OP_DELETE for unacknowledged writes even when OP_MSG is available.  Or is there a separate indirection to turn it into a one-way OP_MSG?

For the (non-legacy) Driver module, what protocol is used to execute unacknowledged writes on server versions 2.6 to 3.4 (inclusive) and for 3.6 to 4.2 (inclusive)?  In particular, is it the same or different as what's done for Driver Legacy?

Reply:
Good point. I totally missed the part about unacknowledged writes.

I'll research the other question about unacknowledged writes in the new API.

Reply:
When using the new API we have support for unacknowledged writes using write commands (no need for OP_DELETE or anything like that).

See this line:

https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs#L111

We only wait for and process a response if one is expected.

Reply:
Note: "CommandMessage" == OP_MSG

Reply:
OK.  Let's leave this alone at least for this ticket.  But let's open a new ticket to remove what should now be dead code if not for the Legacy implementation, which I think could be re-written to use the same write operation as Driver (with some result transformation)

Reply:
From 2.6 to 3.6 the logic is a bit different, but it also doesn't use OP_INSERT, OP_DELETE or OP_UPDATE or any of classes in question (https://github.com/mongodb/mongo-csharp-driver/blob/4f791f7184552d0f87b4bd0b6d062cdde37f7d24/src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingQueryMessageWireProtocol.cs#L143)
Starting from 3.6 (feature "CommandMessage": https://github.com/mongodb/mongo-csharp-driver/blob/4f791f7184552d0f87b4bd0b6d062cdde37f7d24/src/MongoDB.Driver.Core/Core/Misc/Feature.cs#L47) it works as Robert described.
Summing all up, am I assuming correctly that we stick to deprecation instead of deleting for this ticket and create a ticket to completely remove operation classes and update the code which is not in legacy to use `BulkMixedWriteOperation`?

Reply:
I expected all traces of OP_DELETE, OP_INSERT and OP_UPDATE to be removed. They are no longer needed.

Unfortunately we have to keep support for OP_QUERY because the initial isMaster command must be sent using OP_QUERY.